### PR TITLE
Issue1125 st ro be

### LINKED ISSUE #1125: add conditional connectors to StrobeInfoManager and update StROBe component
Conditional connectors should be added to the IDEAS.BoundaryConditions.Occupants.StrobeInfoManager component to be in line with the modelica specification.

The parameters IDEAS.BoundaryConditions.Occupants.StROBe component, which calls the StrobeInfoManager, also needs to be update.  Furthermore, RealExpression `TSet_signal` is adapted as follows:
`TSet_signal(y=TSet_internal+273.15*ones(nZones))`,
since the temperature setpoints given by StROBe are expressed in degrees celcius.

I will make a pull request

### PR DESCRIPTION
**Pull request by Mathadon**
closes #1125

### CODE REVIEW

**Comment by Mathadon** [comment]
@cprotopa do you agree with these changes?

File: IDEAS/BoundaryConditions/Occupants/Extern/StROBe.mo [hunk 1]
```diff
@@ -16,27 +16,30 @@ protected
...
+    y=fV*strobe.tabQCon_internal[id])
     annotation (Placement(transformation(extent={{-120,10},{-160,30}})));
   Modelica.Thermal.HeatTransfer.Sources.PrescribedHeatFlow[nZones] QRad_flow
     annotation (Placement(transformation(extent={{-170,-30},{-190,-10}})));
-  Modelica.Blocks.Sources.RealExpression[nZones] QRad(y=fV*strobe.tabQRad.y[id])
+  Modelica.Blocks.Sources.RealExpression[nZones] QRad(
+    y=fV*strobe.tabQRad_internal[id])
     annotation (Placement(transformation(extent={{-120,-30},{-160,-10}})));
-  Modelica.Blocks.Sources.RealExpression mDHW(y=strobe.tabDHW.y[id]/60)
+  Modelica.Blocks.Sources.RealExpression mDHW(y=strobe.tabDHW_internal[id]/60)
```
Comment (cprotopa) [inline IDEAS/BoundaryConditions/Occupants/Extern/StROBe.mo:27]: Concerning `mDHW` I think a larger update is needed.
StROBe actually outputs liters/minute water withdrawal. Here there is a division by 60 to make it in l/s. We need to see what would be the best units here. Would SI not be m3/s? Also, it's not really "mass flow". A big discussion, I suggest to open in another issue.

**Comment by cprotopa** [comment]
The model works fine for me, so I believe the changes are correctly implemented. 
Of course, the change from Celsius to Kelvin  makes sense. We should just try to inform as many people as possible, because they might have adapted their code elsewhere to perform this change, and this will cause a problem. 

Since we are at it, though, I would suggest we make some more thorough clean up of `IDEAS.BoundaryConditions.Occupants`.  Specifically I think everything in the following folders is totally outdated and not used anywhere, so it could even be removed.
- `IDEAS.BoundaryConditions.Occupants.Extern.Files`
- `IDEAS.BoundaryConditions.Occupants.Extern.Interfaces`
- `IDEAS.BoundaryConditions.Occupants.StROBe`
- `IDEAS.BoundaryConditions.Occupants.Examples.Occupant`

Btw, also:
- `IDEAS.BoundaryConditions.Climate.Time.Elements`

Perhaps this should be done in a separate pull request? What would be best, @Mathadon ?

File: IDEAS/BoundaryConditions/Occupants/Extern/StROBe.mo (hunk 1 above)
Comment (Mathadon) [reply IDEAS/BoundaryConditions/Occupants/Extern/StROBe.mo:27]: SI is kg/s. Best to discuss this in a separate issue indeed, if it's really an issue :)

**Comment by Mathadon** [comment]
Yup, good to make a separate issue/pull request for that!

The C-to-K conversion I did not notice before and is indeed a bit dangerous. @jelgerjansen was this intentional, and what is the rationale for making the change?

**Comment by jelgerjansen** [comment]
@Mathadon I made the change because the other temperatures (eg. the indoor building temperature) are expressed in Kelvin and it made more sense to me to do the conversion at the lowest level. But the manipulation from C-to-K can of course also be done on a higher modeling level.

**Comment by Mathadon** [comment]
That’s true, but as Christina points out, there are many people that are accustomed now to having this variable in C. @cprotopa , I propose to make the conversion but perhaps change the variable name such that users get an error because the older variable no longer exists?

**Comment by cprotopa** [comment]
Hm.. Which variable though? If it's locally changed, say we change the name of `TSet_signal`, nobody will notice, as it's not used elsewhere. 
If, on the other hand, we change the `RealOutput TSet` of the `IDEAS.Templates.Interfaces.BaseClasses.Occupant` to `TSetK`, then it should be also changed in all occupant models in the library that use the template, and in the connection equation in `IDEAS.Templates.Interfaces.Building`. In order for it to be noticed by users, then it should be also changed in `IDEAS.Templates.Interfaces.BaseClasses.HeatingSystem`, and all models that use this latter one.
To me, this seems like the only way to propagate the change of a variable name in all components of the library, making sure that somebody who made a custom component will get an error message.  
Or am I missing something..? 
Of course, they will also probably get some warning of too high temperature here and there. Is this perhaps an approach? Set some max value for the input, that would create a warning? At least we could cover the case where somebody altered the input file to contain K already.  Just a thought...

**Comment by Mathadon** [comment]
Uhm, yes that would definitely be an option. I wasn't sure where the input actually comes from. It's set by a user? Or is it read from file? In both cases we could indeed add a warning/error..

**Comment by cprotopa** [comment]
`StrobeInfoManager` reads external text files, which the user chooses. Normally, these are the ones generated by StROBe, so they are in degrees C. But I can imagine somebody could have chosen to post-process these files to be in K.  

In fact,  `TSet` of the Occupant template is actually specified as 
```
 Modelica.Blocks.Interfaces.RealOutput[nZones] TSet(
    final quantity="ThermodynamicTemperature",
    unit="K",
    displayUnit="degC",
    min=0) 
```
It is also used as such for example in `IDEAS.Templates.Heating.BaseClasses.HysteresisHeating`, where `TSensor` is directly subtracted from `TSet`. Makes sense to have all temperatures everywhere in SI, of course. 

Another option I can think of, is to actually change the StROBe python code to produce the files in K in the first place. This means we would leave the model in IDEAS as is, and assume input files are appropriately in K.  
We can perhaps additionally use a boolean in the `StROBe` component that would specify whether the input is in C or K, and change it or not. Perhaps here can be a check (based on temperature values?).

**Comment by Mathadon** [comment]
The correct approach indeed seems to be to make the change on the strobe side then and to leave the current implementation as it is. @jelgerjansen can you revert the temperature conversion?

@cprotopa I’ll leave it up to you to decide whether or not you’d like to ‘fix’ strobe?

Thx!

Op 2 apr. 2020, om 16:40 heeft Christina Protopapadaki <notifications@github.com<mailto:notifications@github.com>> het volgende geschreven:

Another option I can think of, is to actually change the StROBe python code to produce the files in K in the first place. This means we would leave the model in IDEAS as is, and assume input files are appropriately in K.

**Comment by cprotopa** [comment]
Ok, then I will change StROBe. 

What can be added for clarity in `IDEAS.BoundaryConditions.Occupants.Extern.StrobeInfoManager` is to include the expected units for the inputs. For example this can be at least added in the string parameters comments. E.g.: 
```
 parameter String FilNam_mDHW = "none.txt"   "File with hot water tapping profiles (in l/min)" ;
 parameter String FilNam_QCon = "none.txt"    "File with (convective) internal heat gain profiles (in W)" ;
 parameter String FilNam_TSet = "none.txt"    "File with (main) space heating setpoint profiles (in K)";
```
The units are W for P,Q, Q_Con, Q_Rad, PHp, and PPv. It's then K for TSet and TSet2, and l/min for mDHW (this last one we will discuss in another issue). 

@jelgerjansen could you also add this? Otherwise I will do it afterwards.

**Comment by jelgerjansen** [comment]
I reverted the temperature conversion and updated the documentation. (@Mathadon will add my last commit to this issue)

**Comment by Mathadon** [comment]
this is ready to merge when unit tests pass

**Pull request merged**